### PR TITLE
Add group: and user: urn support for ACL subjects

### PR DESCRIPTION
**Pull request by gschueler**
This extends the ACL Policy format to allow using URNs
for user or role/group subjects in an exact match.

Using a `by: urn: "user:bob"` clause in the ACLpolicy defines
an exact match for a username "bob".

Using `by: urn: "group:dev"` defines exact match for group "dev".

**Is this a bugfix, or an enhancement? Please describe.**

Enhances ACL policy format to let an exact user or group match occur using the `urn:` clause.

### CODE REVIEW

**Comment by gschueler** [comment]
docs pr https://github.com/rundeck/docs/pull/778

File: rundeck-authz/rundeck-authz-api/src/main/java/com/dtolabs/rundeck/core/authorization/RuleEvaluator.java [hunk 1]
```diff
@@ -150,6 +150,24 @@ private static boolean matchesContexts(
...
+            if (subject.getGroups() != null && rule.getUrn() != null && rule.getUrn().startsWith("group:")) {
+                if (subject
+                        .getGroups()
+                        .stream()
+                        .map(a -> "group:" + a)
+                        .collect(Collectors.toSet())
+                        .contains(rule.getUrn())) {
+                    return true;
+                }
+            }
```
Comment (ltamaster) [inline rundeck-authz/rundeck-authz-api/src/main/java/com/dtolabs/rundeck/core/authorization/RuleEvaluator.java:170]: should this code be added to notBy clause?

Comment (gschueler) [reply rundeck-authz/rundeck-authz-api/src/main/java/com/dtolabs/rundeck/core/authorization/RuleEvaluator.java:170]: good catch

**Review by ltamaster — APPROVED** [review]
tested

**Pull request merged**